### PR TITLE
Ali/spot request type support

### PR DESCRIPTION
**Pull request by Aliasgar16**

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

File: lib/chef/knife/ec2_server_create.rb [hunk 1]
```diff
@@ -310,6 +310,11 @@ class Ec2ServerCreate < Knife
         :description => "The maximum hourly USD price for the instance",
         :default => nil
 
+      option :spot_request_type,
+        :long => "--spot-request-type TYPE",
+        :description => "The Spot Instance request type",
```
Comment (siddheshwar-more) [inline lib/chef/knife/ec2_server_create.rb]: Put a possible values for spot_request_type in description.

File: lib/chef/knife/ec2_server_create.rb [hunk 2]
```diff
@@ -395,6 +400,20 @@ def run
           msg_pair("Spot Request ID", spot_request.id)
           msg_pair("Spot Request Type", spot_request.request_type)
           msg_pair("Spot Price", spot_request.price)
+
+          puts 'Do you want to wait for Spot Instance Request fulfillment? (Y/N) '
+          user_choice = STDIN.gets.chomp
```
Comment (siddheshwar-more) [inline lib/chef/knife/ec2_server_create.rb]: You can replace above with https://github.com/chef/knife-ec2/blob/master/lib/chef/knife/ec2_server_delete.rb#L93

**Comment by siddheshwar-more** [comment]
Could you please also add rspec tests for this

**Pull request merged**